Common/Classing/ClassingPlayer.cs

```diff
@@ -0,0 +1,60 @@
+﻿using PathOfTerraria.Common.Looting.VirtualBagUI;
+using PathOfTerraria.Core.UI;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Terraria.Localization;
+using Terraria.ModLoader.IO;
+using Terraria.UI;
+
+namespace PathOfTerraria.Common.Classing;
+
+internal class ClassingPlayer : ModPlayer
+{
+
+	public StarterClasses Class = StarterClasses.None;
+
+	/// <summary>
+	/// Gets the class noun, i.e. Warrior or Marksman. Returns "Stranger" if the player has no selected class.
+	/// </summary>
+	public static string GetClassNoun(Player player, int index = 0)
+	{
+		ClassingPlayer plr = player.GetModPlayer<ClassingPlayer>();
+		string noun;
+
+		if (plr.Class == StarterClasses.None)
+		{
+			noun = Language.GetTextValue("Mods.PathOfTerraria.UI.ClassPages.DefaultNouns");
+		}
+		else
+		{
+			noun = plr.Class.GetNouns();
+		}
+
+		string[] nouns = noun.Split(';');
+
+		if (index >= nouns.Length)
+		{
+			PoTMod.Instance.Logger.Error("[ClassingPlayer] Index was greater than noun count.", new IndexOutOfRangeException());
+		}
+
+		return noun.Split(';')[index % nouns.Length];
```

Comment:
Remainder operator is weird around negative values.
```suggestion
		return nouns[Math.Abs(index) % nouns.Length];
```